common/model-views.cpp

```diff
@@ -4452,21 +4455,25 @@ namespace rs2
                 bool sw_update_required = updates_profile.retrieve_updates(versions_db_manager::LIBREALSENSE);
                 bool fw_update_required = updates_profile.retrieve_updates(versions_db_manager::FIRMWARE);
 
-                _updates_profile = updates_profile.get_update_profile();
-                updates_model::update_profile_model updates_profile_model(_updates_profile, ctx, this);
-
-                if (sw_update_required || fw_update_required)
+                if (auto update_profile = update_profile_protected.lock())
```

Comment:
There are a couple of weak ptrs here... `updates_model`, `update_profile`... can't we count on just one?

This whole piece of code is a little confusing.